packages/core/src/App/Containers/Redirect/redirect.jsx

```diff
@@ -25,6 +26,22 @@ const Redirect = observer(() => {
         is_mobile,
     } = ui;
 
+    // get data from cookies and populate local storage for clients to be logged in
+    const client_accounts = Cookies.get('client.accounts');
+    const active_loginid = Cookies.get('active_loginid');
+
+    if (client_accounts && active_loginid) {
+        localStorage.setItem('client.accounts', client_accounts);
+        localStorage.setItem('active_loginid', active_loginid);
+
+        const domain = getDomainName();
+
+        // remove cookies after populating local storage
+        Cookies.remove('client.accounts', { domain: `.${domain}` });
+        Cookies.remove('active_loginid', { domain: `.${domain}` });
+        window.location.reload();
```

Comment:
[Q] can we check our current redirect function from OAuth, do we also reload, there may already a function to do a reload/side effect to refresh the states.

Reply:
@amam-deriv I went over the code to look into the same things, however we need to respect some redirects from the query params and before redirection and client actions based on those query params, user needed to logged in. If we trigger `init` method from client store, it's kinda async and we get logged out as those actions are under authentication which eventually clears the storage. 

We can have a look again on this.